app/services/kyc_service.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+class KycService
+  # passport front + selfie OR driver license front and back + selfie IR id card front and back + selfie
+  REQUIRED_DOC_AMOUNT = { 'PASSPORT': 2, 'DRIVERS_LICENSE': 3, 'GOVERNMENT_ID': 3 }.freeze
+
+  class << self
+    def profile_step(profile)
+      user = profile.user
+      profile_label = user.labels.find_by(key: :profile)
+
+      if profile_label.nil? # first profile ever
+        user.labels.create(key: :profile, value: profile.state, scope: :private)
+      else
+        profile_label.update(value: profile.state) # re-submitted profile
+      end
+
+      return if Barong::App.config.kyc_provider == 'local' ||
```

Comment:
As we call `profile_step` from model `after_commit` we need to get rid of situations with not ready for verification `drafted` state or decision update `verified` `rejected` states